docs/python/generators/cpp_parser.py

```
@@ -456,15 +460,22 @@ def parse(self):
 def _parse(input: str, location = None | Path):
     return _Parser(input=input, location=location).parse()
-
 def _scan():
     contents = []
-
+    cache = {}
+    cache_file = Path('.parser_cache.pkl')
+    if cache_file.exists():
+        try:
+            with open(cache_file, 'rb') as f:
+                cache = pickle.load(f)
```

Ruff (0.15.21)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)